presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/MultipleDistinctAggregationToMarkDistinct.java

```diff
@@ -99,6 +100,10 @@ private static boolean hasMixedDistinctAndNonDistincts(AggregationNode aggregati
     @Override
     public Result apply(AggregationNode parent, Captures captures, Context context)
     {
+        if (!SystemSessionProperties.useMarkDistinct(context.getSession())) {
```

Comment:
can you provide some guideline when using `MarkDistinctNode` is better or worse? That way it will be easier to modify this into cost-based rule.